lib/jira/client/index.js

```diff
@@ -152,28 +142,27 @@ module.exports = async (jiraHost, gitHubInstallationId, logger) => {
  * Splits commits in data payload into chunks of 400 and makes separate requests
  * to avoid Jira API limit
  */
-const batchedCommitRequest = async (data, options, instance, installationId) => {
-  const filteredCommits = dedupCommits(data)
+const batchedBulkUpdate = async (data, options, instance, installationId) => {
+  const dedupedCommits = data.commits ? dedupCommits(data.commits) : []
 
-  const commitChunks = []
-  while (filteredCommits.length) {
-    commitChunks.push(filteredCommits.splice(0, 400))
+  // Initialize with an empty chunk of commits so we still process the request if there are no commits in the payload
+  const commitChunks = [[]]
+  while (dedupedCommits.length) {
+    commitChunks.push(dedupedCommits.splice(0, 400))
   }
```

Comment:
Since we always start with an empty commit chunk, does that mean any bulk update that includes even a single commit will make at least two requests to the Jira API? 

If that's the case, this may be a good fit for a `do ... while` block, which will always run at least once.

```javascript
const commitChunks = []
do {
  commitChunks.push(dedupedCommits.splice(0, 400))
} while (dedupedCommits.length)
```

This should handle the "no commits" case without adding an empty array in scenarios where commits are present.

Reply:
Fantastic idea ✨ Just implemented this and it's super clean